index.bs

```diff
@@ -736,6 +736,16 @@ Each {{XRSpace}} has a <dfn for="XRSpace">native origin</dfn> that is tracked by
 
 The [=effective origin=] of an {{XRSpace}} can only be observed in the coordinate system of another {{XRSpace}} as an {{XRPose}}, returned by an {{XRFrame}}'s {{XRFrame/getPose()}} method. The spatial relationship between {{XRSpace}}s MAY change between {{XRFrame}}s.
 
+Each {{XRSpace}} has an <dfn for="XRSpace">internal type</dfn>, used to internally distinguish the various kinds of spaces. This value can be <dfn for="XRSpace">viewer space</dfn>, <dfn for="XRSpace">input target ray</dfn>, <dfn for="XRSpace">input grip</dfn>, or one of the variants of {{XRReferenceSpaceType}}.
+
+An {{XRSpace}} of type [=XRSpace/input target ray=], [=XRSpace/input grip=] must have an associated {{XRInputSource}} <dfn for="XRSpace">input source</dfn>.
+
+The [=native origin=] of an {{XRSpace}} with [=XRSpace/internal type=] of [=XRSpace/viewer space=] is the position and orientation of the [=viewer=].
```

Comment:
For this definition in particular, we should perhaps be more specific and define the "viewer space" origin to explicitly coincide with the midpoint of `XRView` origins that will provide optimal rendering for a given XR device.

If not, we may see some UAs place `XRViewerPose.transform` between the view transforms and some UAs place it at another logical viewer position, such as the back of the neck, top of the head, etc., especially given the following language elsewhere:

> NOTE: The {{XRViewerPose}}'s {{XRPose/transform}} can be used to position graphical representations of the [=viewer=] for spectator views of the scene or multi-user interaction.

Reply:
What about devices with more than one view? (This might be something that is better split out into a different issue, I'm trying to clarify the state of the spec as it currently exists without changing behavior)

Reply:
> define the "viewer space" origin to explicitly coincide with the midpoint of XRView

I'm not sure it's correct to say the viewer space's origin is the midpoint of the views. I think in some native systems it's actually set as the centroid of the logical head, and if you have more than two views it could start to skew it.

Reply:
The key question is what we expect apps to do with "viewer space", and if that usage can handle definitions that vary widely among UAs or underlying XR platforms (e.g. some placing it in the center of the head, others placing it at the base of the neck, others placing it between the eyes, etc.).

For example, one reason to expose "viewer space" is to allow apps to raycast from a view-relative position to render a cursor or target with the user's head or device.

Specifically, the [input explainer](https://github.com/immersive-web/webxr/blob/master/input-explainer.md#targeting-ray-pose) mentions that the `targetRaySpace` for an `XRInputSource` with a `targetRayMode` of 'gaze' will match the pose of `viewerSpace`.

For apps designed only for gaze-based targeting that just use `viewerSpace` directly, they will expect viewer space to match the target ray of gaze-targeted input sources as described - for headsets, pointing forward between the eyes (i.e. between the two `XRView`s.  However, if one UA starts its `viewerSpace` ray between the eyes and another UA is free to start its `viewerSpace` ray in the center of the user's head, that target ray will sometimes shoot out from the user's nose instead, producing the wrong app behavior.  The app would instead be required to manually average the two `XRView` rays it receives to produce its gaze targeting ray, which I don't believe is what we intended here.

Reply:
This definition will be even more important to nail down across UAs if we proceed with #623's idea of a "viewer" reference space. (which I believe we should!)